### ISSUE TITLE
Get raw response body in case of JSONSyntaxException

### ISSUE THREAD

**Issue by henhal** [koush/ion#697]
Whenever a backend call normally returning JSON fails for some reason, it can be useful to get the raw response as returned from the server, without attempting to parse it as the given `TypeToken` type etc.

For example, let's say a request returns a HTTP 400 and a raw string "Missing mandatory parameter xyz" as the response body. In this case, I can get hold of the status code through `Response.getHeaders().code()`, but I have found no way of getting the raw response body - `Response.getResult()` is obviously typed T and I don't want to GSON deserialize the body in case of an error.

Is there a way to get the raw response body (as a String) using the current API? If not, would it be possible to have the `Response` or `HeadersResponse` class include it?

**Comment by henhal** [comment]
I just realized that if a JsonSyntaxException occurs, Response will be null, so there's no point having access to the raw response from there.

I achieved my goal by decoupling my GSON deserialization from Ion, always letting Ion return String, and manually invoking GSON on my end. This way I can also deserialize a different kind of objects in case of an error (let's say if the server returns an error description with error code, message, detailed message etc as JSON).

It could however be useful to be able to retrieve the raw response directly from the FutureCallback, regardless of whether a response was available or if an exception was thrown.
